app/src/main/java/ch/epfl/skysync/components/ConfirmFlightDetailUI.kt

```diff
@@ -119,6 +125,41 @@ fun ListElementText(text: String, padding: Dp, testTag: String) {
   )
   Spacer(modifier = Modifier.padding(4.dp))
 }
+
+@Composable
+fun HyperLinkText(title: String, value: String, padding: Dp) {
+  val uriHandler = LocalUriHandler.current
+  val googleMapsLink = "https://www.google.com/maps/search/?api=1&query=${value.replace(" ", "+")}"
+  val string = buildAnnotatedString {
+    pushStringAnnotation(tag = "URL", annotation = googleMapsLink)
+    withStyle(
+        style =
+            SpanStyle(
+                color = MaterialTheme.colorScheme.primary,
+                textDecoration = TextDecoration.Underline)) {
+          append(value)
+        }
+    pop()
+  }
+  Text(
+      modifier = Modifier.fillMaxWidth().padding(horizontal = padding).testTag(title + value),
+      text = title,
+      style = MaterialTheme.typography.headlineSmall,
+      color = Color.Black)
+  Spacer(modifier = Modifier.padding(4.dp))
+
+  ClickableText(
+      text = string,
+      modifier = Modifier.padding(horizontal = padding.plus(4.dp)).testTag(title),
+      style = MaterialTheme.typography.bodyLarge,
+      onClick = { offset ->
+        string.getStringAnnotations(tag = "URL", start = offset, end = offset).firstOrNull()?.let {
+            stringAnnotation ->
+          uriHandler.openUri(stringAnnotation.item)
+        }
```

Comment:
Could add error handling:
?: run {
                // Handle the case where no URL is found
            }

Reply:
I think there will always be an url because i created it on line 132:
val googleMapsLink = "https://www.google.com/maps/search/?api=1&query=${value.replace(" ", "+")}"
then i build a string with tag url with it